core/domain/suggestion_services.py

```diff
@@ -335,6 +347,46 @@ def get_all_suggestions_that_can_be_reviewed_by_user(user_id):
              score_categories, user_id)])
 
 
+def get_reviewable_suggestions(user_id, suggestion_type):
+    """Returns a list of suggestions of given suggestion_type which the user
+    can review.
+
+    Args:
+        user_id: str. The ID of the user.
+        suggestion_type: str. The type of the suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions which the given user is allowed
+            to review.
+    """
+    return ([
+        get_suggestion_from_model(s) for s in (
+            suggestion_models.GeneralSuggestionModel
+            .get_in_review_suggestions_of_suggestion_type(
+                suggestion_type, user_id))
+    ])
+
+
+def get_submitted_suggestions(user_id, suggestion_type):
+    """Returns a list of suggestions of given suggestion_type which the user
+    has submitted.
+
+    Args:
+        user_id: str. The ID of the user.
+        suggestion_type: str. The type of the suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions which the given user has
+        submitted.
+    """
+    return ([
+        get_suggestion_from_model(s) for s in (
+            suggestion_models.GeneralSuggestionModel
+            .get_user_created_suggestions_of_suggestion_type(
```

Comment:
We are going to remove the `query_suggestion` method in separate PR and create a specific method as per requirement.

Reply:
Just curious... Why?

Reply:
As we will be trying to create specific functions as per requirements.

Reply:
I get that, but IIRC I had added the query suggestions function to prevent the need to write new functions everytime. Is there a speed issue or a security issue or something like that?

Reply:
The endpoints were open such that one can perform operation which we were not expecting to happen from a given API. We are thinki9ng of building strict API which allows a specific operation from a given endpoint. 

Also, that wasn't reusable for many cases as we don't allow query based on user_id as the other end only know usernames (not the user ID).